src/allmydata/test/web/test_introducer.py

```diff
@@ -222,16 +233,12 @@ def test_json(self):
             0,
         )
 
-        resource = IntroducerRoot(introducer_node)
-        response = json.loads(
-            self.successResultOf(
+        resource = IntroducerRoot(introducer_node) 
+        response = json.loads(succeeded(
                 render(resource, {b"t": [b"json"]}),
-            ),
-        )
-        self.assertEqual(
-            response,
-            {
+            )._matcher.result)
```

Comment:
The convention of an identifier prefixed with `_` indicates something private or subject to change or which is meant to be an implementation detail.  In general, Tahoe code can't use private APIs from other libraries. Specifically, Tahoe can't use this `_matcher` attribute.

The intended way to use `succeeded` is demonstrated by https://github.com/tahoe-lafs/tahoe-lafs/blob/93fbfb61a8335428ad97d592842a5db78d26e30e/src/allmydata/test/test_eliotutil.py#L252-L260

The assertion here could probably be something like:

```
        response = render(resource, {b"t": [b"json"]})
        expected = {
            u"subscription_summary": {"arbitrary": 2},
            u"announcement_summary": {"arbitrary": 1},
        }
        self.assertThat(response, succeeded(AfterPreprocessing(json.loads, Equals(expected))))
```

Reply:
I  thought as much about this being somewhat weird but never knew exactly what interface to use that would pick our the correct props to make the comparison succeed. Thank you!